src/custom-elements/src/components/radio-button/radio-button.tsx

```diff
@@ -0,0 +1,145 @@
+   */
+  @Prop() label: string;
+  /**
+   * To render the radiobutton input with a respective name attribute
+   */
+  @Prop() name: string;
+  /**
+   * To render the radiobutton in checked state
+   */
+  @Prop() checked?: boolean;
+  /**
+   * To disable the radiobutton
+   */
+  @Prop() disabled?: boolean;
+  /**
+   * To indicate the state
+   */
+  @Prop() state?: ChiStates;
+  /**
+   * To indicate if the element has the focus
+   */
+  @Prop() hasFocus?: boolean;
+  /**
+   * To indicate if info icon should be displayed.
+   */
+  @Prop({ reflect: true }) infoIcon = false;
+  /**
+   * To provide message for info icon popover.
+   */
+  @Prop({ reflect: true }) infoIconMessage = 'Helpful information goes here.';
+
+  private id: string;
+  private labelId: string;
+  private helpButtonId: string;
+  private helpPopoverId: string;
+
+  /**
+   * Triggered when the radio-button has lost the focus
+  */
+ @Event() chiBlur: EventEmitter<string | boolean>;
+ /**
+  * Triggered when the user selects or deselects the radio-button
+  */
+ @Event() chiChange: EventEmitter<string | boolean>;
+  /**
+   * Triggered when the radio-button has the focus
+   */
+  @Event() chiFocus: EventEmitter<string | boolean>;
+
+  connectedCallback() {
+    this.id = this.el.getAttribute('id');
+    addMutationObserver.call(this);
+  }
+
+  componentWillLoad() {
+    this.helpButtonId = `help-button-${uuid4()}`;
+    this.helpPopoverId = `help-popover-${uuid4()}`;
+    this.id = `${uuid4()}-control`;
+    this.labelId = `radio-label-${this.id}`;
+  }
+
+  blur() {
+    this.hasFocus = false;
+    this._emitBlur(this.hasFocus);
+  }
+
+  change(ev: Event) {
+    this.checked = (ev.target as HTMLInputElement).checked;
+    this._emitChange(this.checked);
+  }
+
+  focus() {
+    this.hasFocus = true;
+    this._emitFocus(this.hasFocus);
+  }
+
+  _emitBlur(hasFocus: boolean) {
+    this.chiBlur.emit(hasFocus);
+  }
+
+  _emitChange(checked: boolean) {
+    this.chiChange.emit(checked);
+  }
+
+  _emitFocus(hasFocus: boolean) {
+    this.chiFocus.emit(hasFocus);
+  }
+
+  toggleHelpPopover(): void {
+    const helpPopoverElement = document.getElementById(this.helpPopoverId) as HTMLChiPopoverElement;
+
+    if (helpPopoverElement) {
+      helpPopoverElement.toggle();
+    }
+  }
+
+  render() {
+    const infoIcon = getInfoIcon.call(this);
+    const stateClass = this.state ? `-${this.state}` : '';
+
+    return (
+      <div class={RADIO_CLASSES.RADIO}>
+        <input
+          class={`
+            ${RADIO_CLASSES.INPUT}
+            ${stateClass}
+          `}
+          id={this.id}
+          checked={this.checked}
+          disabled={this.disabled}
+          name={this.name}
+          aria-checked={this.checked}
+          aria-labelledby={this.labelId}
+          onBlur={() => this.blur()}
+          onChange={(ev: Event) => this.change(ev)}
+          onFocus={() => this.focus()}
+          type="radio"
+        />
+        <label
```

Comment:
I would use the `<chi-label>` component here using the properties you need and this way you can delete the changes in the `label.tsx` and delete the new `snippets.tsx` as well.

Reply:
I've tried to use chi-label but selecting an item group doesn't work well. It works when I click on the label but it doesn't when I click in the input. This is the code with chi-label:

```
<input
          class={`
            ${RADIO_CLASSES.INPUT}
            ${stateClass}
          `}
          id={`${this.id}-control`}
          checked={this.checked}
          disabled={this.disabled}
          name={this.name}
          aria-checked={this.checked}
          aria-labelledby={this.labelId}
          onBlur={() => this.blur()}
          onChange={(ev: Event) => this.change(ev)}
          onFocus={() => this.focus()}
          type="radio"
        />
        <chi-label
          id={this.labelId}
          class={`${RADIO_CLASSES.LABEL} ${stateClass}`}
          for={this.id}
          infoIcon={this.infoIcon}
          infoIconMessage={this.infoIconMessage}
        >
          {this.label}
        </chi-label>
```